<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Análises comparativas: evolução, ecologia e conservação • AnaFiloComp</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Análises comparativas: evolução, ecologia e conservação">
<meta property="og:description" content="Este é o reposotório para estudo da disciplina FAP410032-41010066DO/ME (20201)">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">AnaFiloComp</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/cadubio/AnaFiloComp/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">

<div id="análise-filogenética-comparativa" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#an%C3%A1lise-filogen%C3%A9tica-comparativa" class="anchor"></a>Análise Filogenética Comparativa</h1></div>
<div id="disciplina-oferecida-pelo-ppgfap-ufsc" class="section level3">
<h3 class="hasAnchor">
<a href="#disciplina-oferecida-pelo-ppgfap-ufsc" class="anchor"></a>Disciplina oferecida pelo <a href="https://ppgfap.posgrad.ufsc.br/">PPGFAP-UFSC</a>
</h3>
<!-- badges: start -->

<!-- badges: end -->
</div>
</div>
<div id="integração-manipulação-e-visualização-de-dados-filogenéticos" class="section level1">
<h1 class="hasAnchor">
<a href="#integra%C3%A7%C3%A3o-manipula%C3%A7%C3%A3o-e-visualiza%C3%A7%C3%A3o-de-dados-filogen%C3%A9ticos" class="anchor"></a>Integração, manipulação e visualização de dados filogenéticos</h1>
<p>Neste <a href="https://yulab-smu.top/treedata-book/index.html">livro online</a> se encontra um guia para integração, manipulação e visualização de dados filogenéticas através da <a href="https://tidyverse.tidyverse.org/articles/manifesto.html">filosofia <code>tidyverse</code></a>, usando os pacotes <a href="https://www.r-project.org">R</a>: <a href="https://CRAN.R-project.org/package=tidytree">tidytree</a>, <a href="https://doi.org/10.1093/molbev/msz240">treeio</a> e <a href="https://doi.org/10.1111/2041-210X.12628">ggtree</a>.</p>
<p>Os exemplos e tutorias apresentado na disciplina serão convertidos e executados com base neste três pacotes.</p>
<ol>
<li>
<p>Tutoriais/Tutorial filogenias no R-em português(Suzana)/Tutorial phylo R(SAlcantara).txt (as partes de uso básico do <code>R</code> não serão apresentadas aqui)</p>
<ul>
<li>
<p>Lista de pacotes que serão/poderão ser usados na disciplina:</p>
<p><a href="https://cran.r-project.org/web/packages/ade4/index.html" title="ade4">ade4</a>, <a href="https://cran.r-project.org/web/packages/ape/index.html">ape</a>, <a href="https://cran.r-project.org/web/packages/geiger/index.html">geiger</a>, <a href="https://cran.r-project.org/web/packages/diversitree/index.html">diversitree</a>, <a href="https://cran.r-project.org/web/packages/picante/index.html">picante</a>, <a href="https://cran.r-project.org/web/packages/phyloclim/index.html">phyloclim</a>, <a href="https://CRAN.R-project.org/package=vegan">vegan</a>, <a href="https://CRAN.R-project.org/package=foreign">foreign</a>, <a href="https://CRAN.R-project.org/package=nlme">nlme</a>, <a href="https://cran.r-project.org/web/packages/BAMMtools/index.html">BAMMtools</a> e <a href="https://cran.r-project.org/web/packages/phytools/index.html">phytools</a>.</p>
<p>Listas importantes e sempre atualizadas: <a href="https://cran.r-project.org/web/views/Phylogenetics.html">análises filogenéticas comparativas</a></p>
</li>
</ul>
</li>
</ol>
<blockquote>
<p>para arquivos de texto, utilize o programa <a href="https://www.techtudo.com.br/tudo-sobre/textwrangler.html">TextWrangler</a></p>
<p>de preferência, salve os dados originais (planilhas de excel) em formato .csv (sem fórmulas ou caracteres especiais, use . para casa decimais)</p>
</blockquote>
<div id="quarta-lição-aprender-a-manipular-filogenias-em-r" class="section level5">
<h5 class="hasAnchor">
<a href="#quarta-li%C3%A7%C3%A3o-aprender-a-manipular-filogenias-em-r" class="anchor"></a>QUARTA LIÇÃO: aprender a manipular filogenias em R</h5>
<p>comandos úteis: <code>read.tree()</code>, <code>branching.times()</code>, <code>node.depth()</code>, <code>cophenetic.phylo(),</code> <code>vcv()</code>, <code>write.tree()</code>, <code>mrca()</code>, <code>multi2di()</code>, <code>rotate()</code></p>
<p>Plotar: <code>plot.phylo()</code> , <code>add.scale.bar()</code> , <code>identify.phylo()</code> , <code>nodelabels()</code> , <code>subtreeplot()</code> , <code>which.edge()</code> , <code>zoom()</code></p>
</div>
<div id="tutorial-filogenias-intro" class="section level5">
<h5 class="hasAnchor">
<a href="#tutorial-filogenias-intro" class="anchor"></a>Tutorial filogenias-intro:</h5>
</div>
<div id="usando-os-pacotes-indicados-nos-tutoriais" class="section level5">
<h5 class="hasAnchor">
<a href="#usando-os-pacotes-indicados-nos-tutoriais" class="anchor"></a>Usando os pacotes indicados nos tutoriais</h5>
<div class="sourceCode" id="cb1"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true"></a><span class="kw">library</span>(ape)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true"></a><span class="kw">library</span>(geiger)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true"></a><span class="kw">library</span>(phytools)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true"></a><span class="co">#&gt; Loading required package: maps</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true"></a></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true"></a><span class="co">##ler árvore de uma linha (objeto criado: lista de classe phylo):</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true"></a></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true"></a>tt =<span class="st"> "(((((((cow, pig),whale),(bat,(lemur,human))),(robin,iguana)),coelacanth),gold_fish),shark);"</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true"></a></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true"></a>vert.tree &lt;-<span class="st"> </span><span class="kw">read.tree</span>(<span class="dt">text =</span> tt)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true"></a></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true"></a><span class="co">#plotar em diferentes formatos:</span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true"></a><span class="kw">plot</span>(vert.tree)</span></code></pre></div>
<p><img src="reference/figures/README-base-discp-1.png" width="100%"></p>
<div class="sourceCode" id="cb2"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true"></a><span class="co"># Cladograma</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true"></a><span class="kw">plot</span>(vert.tree,<span class="dt">type=</span><span class="st">"cladogram"</span>)</span></code></pre></div>
<p><img src="reference/figures/README-base-discp-2.png" width="100%"></p>
<div class="sourceCode" id="cb3"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true"></a><span class="co"># Sem raiz</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true"></a><span class="kw">plot</span>(<span class="kw">unroot</span>(vert.tree),<span class="dt">type=</span><span class="st">"unrooted"</span>)</span></code></pre></div>
<p><img src="reference/figures/README-base-discp-3.png" width="100%"></p>
<div class="sourceCode" id="cb4"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true"></a><span class="co"># Fan</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true"></a><span class="kw">plot</span>(vert.tree,<span class="dt">type=</span><span class="st">"fan"</span>)</span></code></pre></div>
<p><img src="reference/figures/README-base-discp-4.png" width="100%"></p>
</div>
<div id="usando-os-pacotes-tidy-treeio-e-ggtree" class="section level5">
<h5 class="hasAnchor">
<a href="#usando-os-pacotes-tidy-treeio-e-ggtree" class="anchor"></a>Usando os pacotes ‘tidy’ <code>treeio</code> e <code>ggtree</code>
</h5>
<div class="sourceCode" id="cb5"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true"></a><span class="co"># carregando o pacote 'treeio'</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true"></a><span class="kw">library</span>(treeio)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true"></a><span class="co">#&gt; Registered S3 method overwritten by 'treeio':</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true"></a><span class="co">#&gt;   method     from</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true"></a><span class="co">#&gt;   root.phylo ape</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true"></a><span class="co">#&gt; treeio v1.12.0  For help: https://yulab-smu.github.io/treedata-book/</span></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true"></a><span class="co">#&gt; If you use treeio in published research, please cite:</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true"></a><span class="co">#&gt; LG Wang, TTY Lam, S Xu, Z Dai, L Zhou, T Feng, P Guo, CW Dunn, BR Jones, T Bradley, H Zhu, Y Guan, Y Jiang, G Yu. treeio: an R package for phylogenetic tree input and output with richly annotated and associated data. Molecular Biology and Evolution 2020, 37(2):599-603. doi: 10.1093/molbev/msz240</span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true"></a><span class="co">#&gt; Attaching package: 'treeio'</span></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true"></a><span class="co">#&gt; The following object is masked from 'package:phytools':</span></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true"></a><span class="co">#&gt;     read.newick</span></span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true"></a><span class="co">#&gt; The following object is masked from 'package:geiger':</span></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true"></a><span class="co">#&gt;     treedata</span></span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true"></a><span class="co">#&gt; The following object is masked from 'package:ape':</span></span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true"></a><span class="co">#&gt;     drop.tip</span></span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true"></a></span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true"></a>tt1 &lt;-<span class="st"> "(((((((cow, pig),whale),(bat,(lemur,human))),(robin,iguana)),coelacanth),gold_fish),shark);"</span></span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true"></a></span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true"></a></span>
<span id="cb5-26"><a href="#cb5-26" aria-hidden="true"></a>vert_tree &lt;-<span class="st"> </span><span class="kw">read.newick</span>(<span class="dt">text =</span> tt1)</span>
<span id="cb5-27"><a href="#cb5-27" aria-hidden="true"></a></span>
<span id="cb5-28"><a href="#cb5-28" aria-hidden="true"></a></span>
<span id="cb5-29"><a href="#cb5-29" aria-hidden="true"></a><span class="co">## carregando o pacote 'ggtree'</span></span>
<span id="cb5-30"><a href="#cb5-30" aria-hidden="true"></a><span class="kw">library</span>(ggtree)</span>
<span id="cb5-31"><a href="#cb5-31" aria-hidden="true"></a><span class="co">#&gt; ggtree v2.2.4  For help: https://yulab-smu.github.io/treedata-book/</span></span>
<span id="cb5-32"><a href="#cb5-32" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb5-33"><a href="#cb5-33" aria-hidden="true"></a><span class="co">#&gt; If you use ggtree in published research, please cite the most appropriate paper(s):</span></span>
<span id="cb5-34"><a href="#cb5-34" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb5-35"><a href="#cb5-35" aria-hidden="true"></a><span class="co">#&gt; [36m-[39m Guangchuang Yu. Using ggtree to visualize data on tree-like structures. Current Protocols in Bioinformatics, 2020, 69:e96. doi:10.1002/cpbi.96</span></span>
<span id="cb5-36"><a href="#cb5-36" aria-hidden="true"></a><span class="co">#&gt; [36m-[39m Guangchuang Yu, Tommy Tsan-Yuk Lam, Huachen Zhu, Yi Guan. Two methods for mapping and visualizing associated data on phylogeny using ggtree. Molecular Biology and Evolution 2018, 35(12):3041-3043. doi:10.1093/molbev/msy194</span></span>
<span id="cb5-37"><a href="#cb5-37" aria-hidden="true"></a><span class="co">#&gt; [36m-[39m Guangchuang Yu, David Smith, Huachen Zhu, Yi Guan, Tommy Tsan-Yuk Lam. ggtree: an R package for visualization and annotation of phylogenetic trees with their covariates and other associated data. Methods in Ecology and Evolution 2017, 8(1):28-36. doi:10.1111/2041-210X.12628</span></span>
<span id="cb5-38"><a href="#cb5-38" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb5-39"><a href="#cb5-39" aria-hidden="true"></a><span class="co">#&gt; Attaching package: 'ggtree'</span></span>
<span id="cb5-40"><a href="#cb5-40" aria-hidden="true"></a><span class="co">#&gt; The following object is masked from 'package:ape':</span></span>
<span id="cb5-41"><a href="#cb5-41" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb5-42"><a href="#cb5-42" aria-hidden="true"></a><span class="co">#&gt;     rotate</span></span>
<span id="cb5-43"><a href="#cb5-43" aria-hidden="true"></a></span>
<span id="cb5-44"><a href="#cb5-44" aria-hidden="true"></a><span class="co"># Visualizar</span></span>
<span id="cb5-45"><a href="#cb5-45" aria-hidden="true"></a><span class="kw">ggtree</span>(vert_tree) <span class="op">+</span><span class="st"> </span></span>
<span id="cb5-46"><a href="#cb5-46" aria-hidden="true"></a><span class="st">  </span><span class="kw">geom_tiplab</span>(<span class="dt">size=</span><span class="dv">3</span>, <span class="dt">color=</span><span class="st">"red"</span>)</span></code></pre></div>
<p><img src="reference/figures/README-uso-treio-1.png" width="100%"></p>
<div class="sourceCode" id="cb6"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true"></a><span class="co"># O tipo de árvore é alterado pelo argumento 'layout'</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true"></a><span class="co"># Circular</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true"></a><span class="kw">ggtree</span>(vert_tree, <span class="dt">layout=</span><span class="st">"circular"</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true"></a><span class="st">  </span><span class="kw">geom_tiplab</span>(<span class="dt">size=</span><span class="dv">3</span>, <span class="dt">color=</span><span class="st">"purple"</span>)</span></code></pre></div>
<p><img src="reference/figures/README-uso-treio-2.png" width="100%"></p>
<div class="sourceCode" id="cb7"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true"></a><span class="co"># Inclinada</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true"></a><span class="kw">ggtree</span>(vert_tree, <span class="dt">layout=</span><span class="st">"slanted"</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true"></a><span class="st">  </span><span class="kw">geom_tiplab</span>(<span class="dt">size=</span><span class="dv">3</span>, <span class="dt">color=</span><span class="st">"blue"</span>)</span></code></pre></div>
<p><img src="reference/figures/README-uso-treio-3.png" width="100%"></p>
<div class="sourceCode" id="cb8"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true"></a><span class="co"># Sem raiz</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true"></a><span class="kw">ggtree</span>(vert_tree, <span class="dt">layout=</span><span class="st">"daylight"</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true"></a><span class="st">  </span><span class="kw">geom_tiplab</span>(<span class="dt">size=</span><span class="dv">3</span>, <span class="dt">color=</span><span class="st">"red"</span>)</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true"></a><span class="co">#&gt; Average angle change [1] 0.111110129900635</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true"></a><span class="co">#&gt; Average angle change [2] 0.0369310900693604</span></span></code></pre></div>
<p><img src="reference/figures/README-uso-treio-4.png" width="100%"></p>
<p>objeto de classe phylo tem 4 partes, normalmente “escondidas” (não aparecem quando digitamos o nome do objeto):</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true"></a>vert.tree</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true"></a><span class="co">#&gt; Phylogenetic tree with 11 tips and 10 internal nodes.</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true"></a><span class="co">#&gt; Tip labels:</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true"></a><span class="co">#&gt;   cow, pig, whale, bat, lemur, human, ...</span></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true"></a><span class="co">#&gt; Rooted; no branch lengths.</span></span></code></pre></div>
<p>Independentemente do pacote que o criou:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true"></a>vert_tree</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true"></a><span class="co">#&gt; Phylogenetic tree with 11 tips and 10 internal nodes.</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true"></a><span class="co">#&gt; Tip labels:</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true"></a><span class="co">#&gt;   cow, pig, whale, bat, lemur, human, ...</span></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true"></a><span class="co">#&gt; Rooted; no branch lengths.</span></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true"></a></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true"></a><span class="co">#para ver todas as partes:</span></span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true"></a><span class="kw">str</span>(vert.tree)</span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true"></a><span class="co">#&gt; List of 3</span></span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true"></a><span class="co">#&gt;  $ edge     : int [1:20, 1:2] 12 13 14 15 16 17 18 18 17 16 ...</span></span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true"></a><span class="co">#&gt;  $ Nnode    : int 10</span></span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true"></a><span class="co">#&gt;  $ tip.label: chr [1:11] "cow" "pig" "whale" "bat" ...</span></span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true"></a><span class="co">#&gt;  - attr(*, "class")= chr "phylo"</span></span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true"></a><span class="co">#&gt;  - attr(*, "order")= chr "cladewise"</span></span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true"></a></span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true"></a><span class="kw">str</span>(vert_tree)</span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true"></a><span class="co">#&gt; List of 3</span></span>
<span id="cb10-21"><a href="#cb10-21" aria-hidden="true"></a><span class="co">#&gt;  $ edge     : int [1:20, 1:2] 12 13 14 15 16 17 18 18 17 16 ...</span></span>
<span id="cb10-22"><a href="#cb10-22" aria-hidden="true"></a><span class="co">#&gt;  $ Nnode    : int 10</span></span>
<span id="cb10-23"><a href="#cb10-23" aria-hidden="true"></a><span class="co">#&gt;  $ tip.label: chr [1:11] "cow" "pig" "whale" "bat" ...</span></span>
<span id="cb10-24"><a href="#cb10-24" aria-hidden="true"></a><span class="co">#&gt;  - attr(*, "class")= chr "phylo"</span></span>
<span id="cb10-25"><a href="#cb10-25" aria-hidden="true"></a><span class="co">#&gt;  - attr(*, "order")= chr "cladewise"</span></span></code></pre></div>
<p>outro exemplo com os pacotes do tutorial:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true"></a>tree1 &lt;-<span class="st"> </span><span class="kw">read.tree</span>(<span class="dt">text =</span> <span class="st">"(((A,B),(C,D)),E);"</span>)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true"></a></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true"></a><span class="kw">plot</span>(tree1, <span class="dt">type =</span> <span class="st">"cladogram"</span>, <span class="dt">edge.width =</span> <span class="dv">2</span>)</span></code></pre></div>
<p><img src="reference/figures/README-outro%20exemplo-1.png" width="100%"></p>
<p>mesmo exemplo com os pacotes ‘tidy’:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true"></a>tree2 &lt;-<span class="st"> </span><span class="kw">read.newick</span>(<span class="dt">text =</span> <span class="st">"(((F,G),(H,I)),J);"</span>)</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true"></a></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true"></a><span class="kw">ggtree</span>(tree2, <span class="dt">branch.length=</span><span class="st">'none'</span>) <span class="op">+</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true"></a><span class="st">  </span><span class="kw">geom_tiplab</span>(<span class="dt">size=</span><span class="dv">3</span>, <span class="dt">color=</span><span class="st">"red"</span>)</span></code></pre></div>
<p><img src="reference/figures/README-mesmo%20outro%20exemplo-1.png" width="100%"></p>
<p>objeto phylo com 3 componentes (=explicados a seguir) e uma classe de atributo (=phylo):</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true"></a>tree1<span class="op">$</span>edge</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true"></a><span class="co">#&gt;      [,1] [,2]</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true"></a><span class="co">#&gt; [1,]    6    7</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true"></a><span class="co">#&gt; [2,]    7    8</span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true"></a><span class="co">#&gt; [3,]    8    1</span></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true"></a><span class="co">#&gt; [4,]    8    2</span></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true"></a><span class="co">#&gt; [5,]    7    9</span></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true"></a><span class="co">#&gt; [6,]    9    3</span></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true"></a><span class="co">#&gt; [7,]    9    4</span></span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true"></a><span class="co">#&gt; [8,]    6    5</span></span></code></pre></div>
<p>A matrix edge contém o começo e o final do número do nó para todos os nós e terminais da filogenia.</p>
<p>Por convenção, os terminais da árvore são numerados de 1 ao longo do número n de terminais; e os nós são numerados n:- 1 ao longo do n + m para m nós. m = n - 1 para as árvores totalmente bifurcadas.</p>
<p>O vetor tip.label contém os labels (=identificações) para todos os terminais na árvore.</p>
<p>A ordem dos tip.label é a ordem dos terminais numerados de 1 a n no vetor edge.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true"></a>tree1<span class="op">$</span>tip.label</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true"></a><span class="co">#&gt; [1] "A" "B" "C" "D" "E"</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true"></a></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true"></a>tree2<span class="op">$</span>tip.label</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true"></a><span class="co">#&gt; [1] "F" "G" "H" "I" "J"</span></span></code></pre></div>
<p>O numeral Nnode contém o número de nós internos da árvore, incluindo a raiz da árvore.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true"></a>tree1<span class="op">$</span>Nnode</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true"></a><span class="co">#&gt; [1] 4</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true"></a></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true"></a>tree2<span class="op">$</span>Nnode</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true"></a><span class="co">#&gt; [1] 4</span></span></code></pre></div>
<p>para plotar todas as info:</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true"></a><span class="kw">plot</span>(tree1,<span class="dt">edge.width =</span> <span class="dv">2</span>,<span class="dt">label.offset =</span> <span class="fl">0.1</span>, <span class="dt">type =</span> <span class="st">"cladogram"</span>)</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true"></a><span class="kw">nodelabels</span>()</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true"></a><span class="kw">tiplabels</span>()</span></code></pre></div>
<p><img src="reference/figures/README-rotulos-1.png" width="100%"></p>
<p>para plotar todas as info com os pacotes ‘tidy’:</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true"></a><span class="kw">ggtree</span>(tree2, <span class="dt">branch.length=</span><span class="st">'none'</span>) <span class="op">+</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true"></a><span class="st">  </span><span class="kw">geom_text</span>(<span class="kw">aes</span>(<span class="dt">label =</span> node), <span class="dt">color =</span> <span class="st">"blue"</span>, <span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">hjust =</span> <span class="fl">-.3</span>) <span class="op">+</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true"></a><span class="st">  </span><span class="kw">geom_tiplab</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">geom =</span> <span class="st">"label"</span>, <span class="dt">color=</span><span class="st">"red"</span>, <span class="dt">hjust =</span> <span class="fl">-.5</span>)</span></code></pre></div>
<p><img src="reference/figures/README-rotulos%20tidy-1.png" width="100%"></p>
<p>“phylo” tb pode ter outros componentes, os mais comuns são <code>edge.length</code> (um vetor de classe <code>numeric</code> contendo todos os comprimentos dos ramos da árvore na mesma ordem em que as linhas no vetor edge; e <code>root.edge</code>, um valor numérico dando o comprimento do ramo da raiz, se enraizada.</p>
<p>Outros elementos e atributos podem ser adicionados para tipos especiais de árvores filogenéticas.</p>
<p>Escrevendo e lendo árvores filogenéticas:</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true"></a><span class="kw">write.tree</span>(tree1, <span class="st">"example1.tre"</span>)</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true"></a></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true"></a><span class="kw">cat</span>(<span class="kw">readLines</span>(<span class="st">"example1.tre"</span>))</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true"></a><span class="co">#&gt; (((A,B),(C,D)),E);</span></span></code></pre></div>
<p>utilizando phytools:</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true"></a><span class="kw">writeNexus</span>(tree1,<span class="st">"example1.nex"</span>)</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true"></a></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true"></a><span class="kw">cat</span>(<span class="kw">readLines</span>(<span class="st">"example1.nex"</span>),<span class="dt">sep=</span><span class="st">"</span><span class="ch">\n</span><span class="st">"</span>)</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true"></a><span class="co">#&gt; #NEXUS</span></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true"></a><span class="co">#&gt; [R-package PHYTOOLS, Thu Oct  8 19:18:08 2020]</span></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true"></a><span class="co">#&gt; </span><span class="re">BEGIN</span><span class="co"> TAXA;</span></span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true"></a><span class="co">#&gt;  DIMENSIONS NTAX = 5;</span></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true"></a><span class="co">#&gt;  TAXLABELS</span></span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true"></a><span class="co">#&gt;      A</span></span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true"></a><span class="co">#&gt;      B</span></span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true"></a><span class="co">#&gt;      C</span></span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true"></a><span class="co">#&gt;      D</span></span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true"></a><span class="co">#&gt;      E</span></span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true"></a><span class="co">#&gt;  ;</span></span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true"></a><span class="co">#&gt; </span><span class="re">END</span><span class="co">;</span></span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true"></a><span class="co">#&gt; </span><span class="re">BEGIN</span><span class="co"> TREES;</span></span>
<span id="cb19-18"><a href="#cb19-18" aria-hidden="true"></a><span class="co">#&gt;  TRANSLATE</span></span>
<span id="cb19-19"><a href="#cb19-19" aria-hidden="true"></a><span class="co">#&gt;      1   A,</span></span>
<span id="cb19-20"><a href="#cb19-20" aria-hidden="true"></a><span class="co">#&gt;      2   B,</span></span>
<span id="cb19-21"><a href="#cb19-21" aria-hidden="true"></a><span class="co">#&gt;      3   C,</span></span>
<span id="cb19-22"><a href="#cb19-22" aria-hidden="true"></a><span class="co">#&gt;      4   D,</span></span>
<span id="cb19-23"><a href="#cb19-23" aria-hidden="true"></a><span class="co">#&gt;      5   E</span></span>
<span id="cb19-24"><a href="#cb19-24" aria-hidden="true"></a><span class="co">#&gt;  ;</span></span>
<span id="cb19-25"><a href="#cb19-25" aria-hidden="true"></a><span class="co">#&gt;  TREE * UNTITLED = [&amp;R] (((1,2),(3,4)),5);</span></span>
<span id="cb19-26"><a href="#cb19-26" aria-hidden="true"></a><span class="co">#&gt; </span><span class="re">END</span><span class="co">;</span></span></code></pre></div>
<p>Simulando, plotando, extraindo clados &amp; cortando tips (terminais) das árvores:</p>
<p>(escolher o valor de seed para garantir repetibilidade)</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true"></a><span class="kw">set.seed</span>(<span class="dv">1</span>)</span></code></pre></div>
<p>simular uma árvore com modelo birth-death utilizando phytools</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true"></a>tree &lt;-<span class="st"> </span><span class="kw">pbtree</span>(<span class="dt">b =</span> <span class="dv">1</span>, <span class="dt">d =</span> <span class="fl">0.2</span>, <span class="dt">n =</span> <span class="dv">40</span>)</span></code></pre></div>
<p>critério de parada é 40 espécies viventes, neste caso:</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true"></a><span class="kw">plotTree</span>(tree)</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true"></a><span class="kw">nodelabels</span>()</span></code></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-10-1.png" width="100%"></p>
<p>Utilizando os pacotes ‘tidy’</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true"></a><span class="kw">ggtree</span>(tree) <span class="op">+</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true"></a><span class="st">  </span><span class="kw">geom_nodelab</span>(<span class="kw">aes</span>(<span class="dt">label =</span> node), <span class="dt">colour =</span> <span class="st">"blue"</span>, <span class="dt">size =</span> <span class="dv">2</span>, <span class="dt">hjust =</span> <span class="fl">.2</span>) <span class="op">+</span></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true"></a><span class="st">  </span><span class="kw">geom_tiplab</span>(<span class="dt">colour =</span> <span class="st">"red"</span>, <span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">hjust =</span> <span class="fl">-.3</span>)</span></code></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-11-1.png" width="100%"></p>
<p>ok, agora extraia o clado descendente a partir do nó de número 62</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true"></a>tt62 &lt;-<span class="st"> </span><span class="kw">extract.clade</span>(tree, <span class="dv">62</span>)</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true"></a> </span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true"></a><span class="kw">plotTree</span>(tt62)</span></code></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-12-1.png" width="100%"></p>
<p>Utilizando os pacotes ‘tidy’:</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true"></a><span class="co"># Note que o pacote `ggtree` possui a função `viewClade`, que mostra um clado</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true"></a><span class="co"># da árvore criada "tree" sem a necessidade de se extrair um subconjunto (tt62)</span></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true"></a><span class="co"># da árvore para então proceder a visualização. </span></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true"></a></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true"></a>p &lt;-<span class="st"> </span><span class="kw">ggtree</span>(tree) <span class="op">+</span><span class="st"> </span><span class="kw">geom_tiplab</span>()</span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true"></a></span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true"></a><span class="kw">viewClade</span>(p, <span class="dt">node =</span> <span class="dv">62</span>)</span></code></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-13-1.png" width="100%"></p>
</div>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Browse source code at <br><a href="https://github.com/cadubio/AnaFiloComp/">https://​github.com/​cadubio/​AnaFiloComp/​</a>
</li>
<li>Report a bug at <br><a href="https://github.com/cadubio/AnaFiloComp/issues">https://​github.com/​cadubio/​AnaFiloComp/​issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Carlos Eduardo Siqueira <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0003-4232-8082" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

  <div class="dev-status">
<h2>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://yulab-smu.top/treedata-book/chapter1.html#pc"><img src="https://img.shields.io/badge/taxonomy-trees-green.svg" alt="Taxonomy: trees"></a></li>
</ul>
</div>
</div>
</div>


      <footer><div class="copyright">
  <p>Developed by Carlos Eduardo Siqueira.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
